File: tmp/upvote-tifu.py
import asyncio
from playwright.async_api import async_playwright

async def main():
    cdp_url = 'ws://127.0.0.1:53900/devtools/browser/83576bd8-0947-4222-ad94-8c03564228e4'
    target_url = 'https://www.reddit.com/r/tifu/comments/1s099j8/tifu_by_building_an_app_that_notifies_me_when_my/'

    async with async_playwright() as p:
        browser = await p.chromium.connect_over_cdp(cdp_url)

        contexts = browser.contexts
        if contexts:
            pages = contexts[0].pages
            if pages:
                page = pages[0]
                # Close extra tabs
                for extra in pages[1:]:
                    await extra.close()
            else:
                page = await contexts[0].new_page()
        else:
            ctx = await browser.new_context()
            page = await ctx.new_page()

        print('Navigating to target URL...')
        await page.goto(target_url, wait_until='domcontentloaded')
        await page.wait_for_timeout(3000)

        upvote_btn = page.get_by_role('button', name='upvote')
        count = await upvote_btn.count()
        print(f'Upvote button count: {count}')

        if count == 0:
            print('ERROR: Upvote button not found')
            return

        aria_before = await upvote_btn.first.get_attribute('aria-pressed')
        print(f'aria-pressed before: {aria_before}')

        if aria_before == 'true':
            print('ALREADY_UPVOTED: Post was already upvoted')
            return

        print('Clicking upvote button...')
        await upvote_btn.first.click()
        await page.wait_for_timeout(1500)

        aria_after = await upvote_btn.first.get_attribute('aria-pressed')
        print(f'aria-pressed after: {aria_after}')

        if aria_after == 'true':
            print('SUCCESS: Post upvoted successfully')
        else:
            print(f'FAILURE: aria-pressed is still "{aria_after}" after clicking')

asyncio.run(main())

File: tmp/upvote-tifu.js
const { chromium } = require('playwright');

(async () => {
  const cdpUrl = 'ws://127.0.0.1:53683/devtools/browser/5aaa5095-79ed-4595-a7e9-ec92573a04b4';
  const targetUrl = 'https://www.reddit.com/r/tifu/comments/1s099j8/tifu_by_building_an_app_that_notifies_me_when_my/';

  const browser = await chromium.connectOverCDP(cdpUrl);
  const contexts = browser.contexts();
  const context = contexts[0];

  // Get all pages, close extras, keep one
  let pages = context.pages();
  console.log(`Found ${pages.length} tab(s)`);

  let page;
  if (pages.length === 0) {
    page = await context.newPage();
  } else {
    page = pages[0];
    for (let i = 1; i < pages.length; i++) {
      await pages[i].close();
    }
  }

  console.log('Navigating to target URL...');
  await page.goto(targetUrl, { waitUntil: 'domcontentloaded', timeout: 30000 });

  // Wait for JS hydration
  await page.waitForTimeout(3000);
  console.log('Page loaded, looking for upvote button...');

  // Find upvote button
  const upvoteBtn = page.getByRole('button', { name: /^upvote$/i }).first();
  
  let btnExists = false;
  try {
    await upvoteBtn.waitFor({ timeout: 10000 });
    btnExists = true;
  } catch (e) {
    console.log('Upvote button not found via role, trying alternative selectors...');
  }

  if (!btnExists) {
    // Try alternative
    const altBtn = page.locator('[aria-label="upvote"]').first();
    try {
      await altBtn.waitFor({ timeout: 5000 });
      const ariaPressed = await altBtn.getAttribute('aria-pressed');
      console.log(`Alt button found, aria-pressed: ${ariaPressed}`);
      if (ariaPressed === 'true') {
        console.log('RESULT: Already upvoted');
        await browser.close();
        return;
      }
      await altBtn.click();
      await page.waitForTimeout(1500);
      const newAriaPressed = await altBtn.getAttribute('aria-pressed');
      console.log(`After click, aria-pressed: ${newAriaPressed}`);
      if (newAriaPressed === 'true') {
        console.log('RESULT: SUCCESS - upvote confirmed');
      } else {
        console.log('RESULT: FAILED - aria-pressed did not change to true');
      }
      return;
    } catch (e2) {
      console.log('Alternative selector also failed:', e2.message);
      process.exit(1);
    }
  }

  const ariaPressed = await upvoteBtn.getAttribute('aria-pressed');
  console.log(`Upvote button found, aria-pressed: ${ariaPressed}`);

  if (ariaPressed === 'true') {
    console.log('RESULT: Already upvoted — no action taken');
    return;
  }

  console.log('Clicking upvote button...');
  await upvoteBtn.click();
  await page.waitForTimeout(1500);

  const newAriaPressed = await upvoteBtn.getAttribute('aria-pressed');
  console.log(`After click, aria-pressed: ${newAriaPressed}`);

  if (newAriaPressed === 'true') {
    console.log('RESULT: SUCCESS - upvote confirmed');
  } else {
    console.log(`RESULT: FAILED - aria-pressed is "${newAriaPressed}" after click`);
  }
})().catch(err => {
  console.error('ERROR:', err.message);
  process.exit(1);
});
